### PR TITLE
Introduce terraform lock and add deletion protection flag to google container cluster resource

### PR DESCRIPTION
**Pull request by petechd**
### What is the context of this PR?
Since we started using version 5.0.0+ of the google provider, we must explicitly set deletion_protection=false in order to destroy a cluster. Otherwise, our destroy infrastructure tasks will fail. We are also pinning provider version and introducing terraform lock file so this won't happen in the future.

### How to review
Run baseline benchmark and check if it's failing, use this branch of load generator in reflown benchmark pipeline and check if all tasks pass. You might have to fly new pipeline when using this branch if you don't want to manually edit tfstate file after the first run.

### CODE REVIEW

**Comment by petechd** [comment]
[Investigate daily benchmark failures](https://trello.com/c/WUCUB2mT/5907-investigate-daily-benchmark-failures)

**Comment by petechd** [comment]
[Introduce terraform lock file for the eq-terraform-load-generator](https://trello.com/c/jPL5wiAh/5908-introduce-terraform-lock-file-for-the-eq-terraform-load-generator)

File: versions.tf [hunk 1]
```diff
@@ -1,4 +1,9 @@
-
 terraform {
   required_version = ">= 1.3.7"
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = ">= 5.0.0, < 6.0.0"
```
Comment (berroar) [inline versions.tf:6]: How come < 6, and not just `>= 5.0.0`? Is there a version 6?

**Review by katie-gardner — APPROVED** [review]
Fixes the destroy step on my personal benchmarking pipeline and changes look sound 👍

**Review by berroar — APPROVED** [review]
Tested benchmark and running with branch, destroy step works as expected with the changes 👍

**Pull request merged**